Fix crash in TSupergroups

fingolfin opened the pull request:
NameSstd was not initialized, in particular NameSstd.trafo was not.
But we call free_CARATname_TYP(NameSstd), which does this, with
Name = NameSstd:

    if (Name.trafo)
       free_mat(Name.trafo);

Now Name.trafo had a random value, usually not zero; trying to free
it then causes a crash.

Fixes #8

gaehler commented:
The added line should rather read:
`memset(&NameSstd, 0, sizeof(NameSstd));`
In `tst`, you can run a (still commented out) test::
`../bin/carat/TSupergroups -t Ex14_R`
The file `IT88.txt` from  #8 is unfortunately wrong (those generators are acting from the right instead of from the left). `TSupergroups` runs through, but the result is nonsense.

fingolfin commented:
Fixed. Indeed, that was nonsense, thanks for catching it (I had a different fix first, but it required a C99 extension, so I rewrote it and was careless *sigh*).

fingolfin commented:
On Linux, new pages (allocated by the memory manager) are by default zero initialized. So is the stack, at least initially. This is why very often, code which fails to explicitly init variables, works on Linux "by chance", as those uninitialize variables end up being 0, which often is handled right by code.

But on other OSes, including macOS and Windows, things are quite different, and so it is much, much more likely that such bad code crashes. I've observed this countless times in the past.
